analytics/state_checker.go

```diff
@@ -23,5 +26,9 @@ func NewStateChecker(repository env.Repository) StateChecker {
 
 // Enabled ...
 func (s stateChecker) Enabled() bool {
+	if s.envRepository.Get(V2DisabledEnvKey) == "true" {
```

Comment:
We could check if it is set, and always use its value, if it is. This would make it less confusing.

Reply:
Do you mean treating any value as a "true" value and only treat an unset env var as "false"? I would rather keep this new flag consistent with the existing one (which is only activated if its value is "true")